gui/src/system/SystemPages.tsx

```diff
@@ -374,19 +386,31 @@ async function submitConfig(edited: boolean) {
       null : JSON.parse(mappingvalvenormcloseds[i].value.toLowerCase())
   }
   console.log(entries);
-  const success = await sendConfig(serverIp() as string, {id: configName, mappings: entries} as Config) as object;
+  const response = await sendConfig(serverIp() as string, {id: configName, mappings: entries} as Config) as Response;
+  const success = response as object;
   const statusCode = success['status' as keyof typeof success];
   if (statusCode != 200) {
     refreshConfigs();
+    if (statusCode == 400) {
+      setCurrentConfigurationErrorCode('ERROR : BAD REQUEST');
+    } else if (statusCode == 418) {
+      setCurrentConfigurationErrorCode("ERROR : I'M A TEAPOT");
+    } else {
+      setCurrentConfigurationErrorCode("ERROR CODE " + (statusCode as String));
```

Comment:
Casting is both not necessary and invalid here. Type casting is stripped when transpiled to JS; it's just a hint to the TS compiler. This does not perform a conversion to a `String` as expected. It just tells TypeScript "trust me bro, this is a string." Just remove the cast, it's fine to add a `number` to a `string`.